src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs

```diff
@@ -50,83 +52,172 @@ static IdentityModelEventSource()
         /// </summary>
         public static IdentityModelEventSource Logger { get; }
 
+        [Event(6, Level = EventLevel.LogAlways)]
+        public void WriteAlways(string message)
+        {
+            if (IsEnabled())
+            {
+                message = PrepareMessage(EventLevel.LogAlways, message);
+                WriteEvent(6, message);
+            }
+        }
+
+        [NonEvent]
+        public void WriteAlways(string message, params object[] args)
+        {
+            if (IsEnabled())
+            {
+                if (args != null)
+                    WriteAlways(string.Format(CultureInfo.InvariantCulture, message, args));
+                else
+                    WriteAlways(message);
+            }
+        }
+
         [Event(1, Level = EventLevel.Verbose)]
         public void WriteVerbose(string message)
         {
-            if (IsEnabled() && _logLevel >= EventLevel.Verbose)
+            if (IsEnabled() && LogLevel >= EventLevel.Verbose)
             {
-                message = PrepareMessage(message, EventLevel.Verbose);
+                message = PrepareMessage(EventLevel.Verbose, message);
                 WriteEvent(1, message);
             }
         }
 
+        [NonEvent]
+        public void WriteVerbose(string message, params object[] args)
```

Comment:
I don't think we need 2 overrides of this method. We can modify the existing method's signature to (string message, params object[] args). That way, we can call the method with or without args. All this variant is doing is formatting which we are doing in PrepareMessage already. We can have something like this:
        [Event(3, Level = EventLevel.Warning)]
        public void WriteWarning(string message, params object[] args)
        {
            if (IsEnabled() && LogLevel >= EventLevel.Warning)
            {
                message = PrepareMessage(EventLevel.Warning, message, args);
                WriteEvent(3, message);
            }
        }